src/lib/up-storage.js

```diff
@@ -540,7 +540,7 @@ class ProxyStorage implements IProxy {
       //
       if (this.proxy === false) {
         headers['X-Forwarded-For'] = (
-          req && req.headers['x-forwarded-for']
+          req.headers['x-forwarded-for']
```

Comment:
Actually can be undefined 

```
// line 447
const readStream = this.request({
      uri_full: url,
      encoding: null,
      headers: {
        Accept: contentTypeAccept,
      },
    });
```

Reply:
See https://github.com/DanielRuf/verdaccio/blob/46787f3306da87684ab17d2452318c0108d14010/src/lib/up-storage.js#L534 which already checks it or did I oversee something?

Reply:
I did not see that 👍  My bad.

Reply:
Thanks @DanielRuf ! You are on 🔥 today.